GUI/Types/Exporter/GltfModelExporter.cs

```diff
@@ -130,16 +136,102 @@ private Mesh CreateGltfMesh(string meshName, VMesh vmesh, ModelRoot model)
                     primitive.WithIndicesAccessor(PrimitiveType.TRIANGLES, indices);
 
                     // Add material
-                    //var primitive = mesh.CreatePrimitive()
-                    //    .WithVertexAccessor("POSITION", positions)
-                    //    .WithIndicesAccessor(PrimitiveType.TRIANGLES, indices);
-                    //.WithMaterial(material);
+                    var materialPath = drawCall.GetProperty<string>("m_material");
+
+                    var renderMaterial = context.MaterialLoader.GetMaterial(materialPath);
+
+                    var materialNameTrimmed = Path.GetFileNameWithoutExtension(materialPath);
+                    var bestMaterial = GenerateGLTFMaterialFromRenderMaterial(renderMaterial, model, context, materialNameTrimmed);
+                    primitive.WithMaterial(bestMaterial);
                 }
             }
 
             return mesh;
         }
 
+        private Material GenerateGLTFMaterialFromRenderMaterial(RenderMaterial renderMaterial, ModelRoot model, VrfGuiContext context, string materialName)
+        {
+            var material = model
+                    .CreateMaterial(materialName + "_material")
+                    .WithPBRMetallicRoughness();
+            material.Alpha = AlphaMode.MASK;
+
+            foreach (var renderTexture in renderMaterial.Material.TextureParams)
+            {
+                var texturePath = renderTexture.Value;
+
+                var fileName = Path.GetFileNameWithoutExtension(texturePath);
+
+                Console.WriteLine($"Exporting texture for mesh: {texturePath}");
+                var textureResource = context.LoadFileByAnyMeansNecessary(texturePath + "_c");
+
+                if (textureResource == null)
+                {
+                    continue;
+                }
+                var bitmap = ((ValveResourceFormat.ResourceTypes.Texture)textureResource.DataBlock).GenerateBitmap();
+
+                if (fileName.Contains("color"))
+                {
+                    // expensive transparency workaround for color maps 
+                    for (int row = 0; row < bitmap.Width; row++) {
+                        for (int col = 0; col < bitmap.Height; col++)
+                        {
+                            var pixelAt = bitmap.GetPixel(row, col);
+                            bitmap.SetPixel(col, row, new SKColor(pixelAt.Red, pixelAt.Green, pixelAt.Blue,  255));
+                        }
+                    }
+                }
+                var textureImage = SKImage.FromBitmap(bitmap);
+                using var data = textureImage.Encode(SKEncodedImageFormat.Png, 100);
+
+                var image = model.UseImageWithContent(data.ToArray());
+                // TODO find a way to change the image's URI to be the image name, right now it turns into (model)_0, (model)_1....
```

Comment:
Is this comment still accurate? Seems okay to use the fileName to me, it might just happen to have the same name as the model ofcourse...

Reply:
You can't rename the image's uri, so for a file gordon.gltf the associated images will be gordon_0.png, gordon_1.png... etc

This will likely be fixed by writing a post-export script